### PR TITLE
feat: OPTIC-1337: Collect usage metrics internally

### PR DESCRIPTION
**Pull request by bmartel**
### PR fulfills these requirements
- [X] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [X] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [X] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
_(check all that apply)_
- [X] Product
- [ ] Backend (Database)
- [ ] Backend (API)
- [X] Frontend

### Describe the reason for change
To be able to better understand product usage within the application itself. Adheres to `COLLECT_ANALYTICS` setting and will be a noop in the cases that setting is falsey.

### Which logical domain(s) does this change affect?
Product Metrics

### CODE REVIEW

File: web/libs/core/project.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "core",
+  "$schema": "../../node_modules/nx/schemas/project-schema.json",
+  "sourceRoot": "libs/core/src",
+  "projectType": "library",
+  "tags": [],
+  "targets": {
+    "lint": {
+      "executor": "@nx/eslint:lint",
```
Comment (yyassi-heartex) [inline web/libs/core/project.json]: I see eslint here - was that intended? would be awesome to just use biome through out if possible

File: web/libs/core/src/lib/utils/analytics.test.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+    });
+
+    // Reset APP_SETTINGS before each test
+    (window as any).APP_SETTINGS = {
+      collect_analytics: true,
+    };
+
+    // Mock window.location
+    Object.defineProperty(window, "location", {
+      value: { href: "http://test.com/page" },
```
Comment (yyassi-heartex) [inline web/libs/core/src/lib/utils/analytics.test.ts]: might be good to define `"http://test.com/page"` as a variable and reuse it as opposed to having the string in a number of places

**Review by yyassi-heartex — APPROVED** [review]
overall looks good - just mildly concerned that eslint is being used again instead of biome - and a minor tweak to one of the tests

File: web/libs/core/project.json (hunk 1 above)
Comment (bmartel) [reply web/libs/core/project.json]: ahh darn missed this removal, thanks for catching it!

Comment (bmartel) [reply web/libs/core/project.json]: The generator is still including eslint + prettier by default which it actually should be configured to omit this for any generated libs.

**Review by hlomzik — APPROVED** [review]

**Review by AndrejOros — APPROVED** [review]

**Pull request merged**